Packs/ProofpointThreatResponse/Integrations/ProofpointThreatResponse/ProofpointThreatResponse.py

```diff
@@ -452,26 +452,49 @@ def get_incidents_request(params):
 def get_incidents_batch_by_time_request(params):
     """Perform an API request to get incidents from ProofPoint in batches to prevent a timeout.
 
+    As the api does not return the results in an specific order, we query the api on specific time frames using
+    created_before and created_after using the fetch delta parameter.
     Args:
         params(dict): The params of the request
 
     Returns:
         list. The incidents returned from the API call
     """
     incidents_list = []  # type:list
-    time_delta = timedelta(hours=6)  # batch by days
+    new_fetched_incidents = []
+
+    fetch_delta = int(params.get('fetch_delta', '6'))
+    fetch_limit = int(params.get('fetch_limit', '50'))
+    already_fetched = params.get('already_fetched', [])
+    last_fetch = params.get('created_after')
 
     current_time = datetime.now()
+    time_delta = timedelta(hours=fetch_delta)  # batch by days
     created_after = datetime.strptime(params.get('created_after'), TIME_FORMAT)
     created_before = created_after + time_delta
+
     request_params = {
         'state': params.get('state'),
         'created_after': created_after.isoformat().split('.')[0] + 'Z',
         'created_before': created_before.isoformat().split('.')[0] + 'Z'
     }
 
-    while created_before < current_time:
-        incidents_list.extend(get_incidents_request(request_params))
+    while created_before < current_time and len(incidents_list) < fetch_limit:
+        incidents = get_incidents_request(request_params)
+        filtered_incidents_list = filter_incidents(incidents)
+
+        for incident in filtered_incidents_list:
+            # if reached to fetch limit, no need to continue to go through the incidents
+            if len(incidents_list) >= fetch_limit:
+                break
+            # only if incident was not fetched already add it to the incidents list
+            if incident.get('id') not in already_fetched:
+                incidents_list.append(incident)
+                new_fetched_incidents.append(incident.get('id'))
+
+        # save the last fetch for the next fetch before advancing it
+        # We would like to run again on the last batch to add the incident that might not been added due to the limit
+        last_fetch = created_after.isoformat().split('.')[0] + 'Z'
```

Comment:
don't you have it in the main fetch func already?

Reply:
If I understand correctly your question , then No,
In the main function the previous code took the last run from the last incident using `incidents[-1]['occured']`
However, as the incidents are not ordered in a certain order this usage will cause duplicate incidents/ missing incidents.

As a result I am saving each api call the `created_after` value and will set this value as the lastRunTime.
This will cause the next fetch iteration to run on the same incidents but for those who werent added due to the limit they will be added the next time and to those who were already added wont cause duplicated due to the `already_fetched`

Reply:
how does this value ever move forward?
this is to make sure that the response does not get too big.

Reply:
each iteration of the while loop the created_after is being moved